jni/src/nmslib_wrapper.cpp

```diff
@@ -157,12 +157,6 @@ void knn_jni::nmslib_wrapper::CreateIndex(knn_jni::JNIUtilInterface * jniUtil, J
         }
         jniUtil->ReleaseIntArrayElements(env, idsJ, idsCpp, JNI_ABORT);
 
-        // Releasing the vectorsAddressJ memory as that is not required once we have created the index.
-        // This is not the ideal approach, please refer this gh issue for long term solution:
-        // https://github.com/opensearch-project/k-NN/issues/1600
-        //commons::freeVectorData(vectorsAddressJ);
-        delete inputVectors;
```

Comment:
if you don't release the memory here then during the createIndex and save index the memory will shoot up. Hence this was added specially here to avoid the memory shoots ups. Lets not remove this.

Reply:
https://github.com/opensearch-project/k-NN/pull/1950#discussion_r1721170735